### PR TITLE
Added support for proc and mem to perform sequentially

### PR DESCRIPTION
**Pull request by shirishaganta1**
Proc: Added support which do add and remove procs
based on the availability of system resources
Mem: added support which add memory quantity as per the lmb based on the availability of system resources

### CODE REVIEW

**Comment by shirishaganta1** [comment]
JOB ID     : db8c1b0e45884ef0142412165d7ca25dffca669b
JOB LOG    : /home/geet/avocado-fvt-wrapper/results/job-2024-04-28T04.05-db8c1b0/job.log
 (1/3) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-39fa: STARTED
 (1/3) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-39fa:  PASS (248.49 s)
 (2/3) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-39fa: STARTED
 (2/3) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-39fa:  PASS (126.26 s)
 (3/3) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-39fa: STARTED
 (3/3) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-39fa:  PASS (564.89 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/geet/avocado-fvt-wrapper/results/job-2024-04-28T04.05-db8c1b0/results.html
JOB TIME   : 984.97 s
[cpu_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15141791/cpu_add_run-config-lpar_mode-dedicated.log)
[cpu_rm_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15141794/cpu_rm_run-config-lpar_mode-dedicated.log)
[mem_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15141797/mem_add_run-config-lpar_mode-dedicated.log)

**Comment by shirishaganta1** [comment]
JOB ID     : db8c1b0e45884ef0142412165d7ca25dffca669b
JOB LOG    : /home/geet/avocado-fvt-wrapper/results/job-2024-04-28T04.05-db8c1b0/job.log
 (1/3) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-39fa: STARTED
 (1/3) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-39fa:  PASS (248.49 s)
 (2/3) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-39fa: STARTED
 (2/3) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-39fa:  PASS (126.26 s)
 (3/3) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-39fa: STARTED
 (3/3) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-39fa:  PASS (564.89 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/geet/avocado-fvt-wrapper/results/job-2024-04-28T04.05-db8c1b0/results.html
JOB TIME   : 984.97 s
[cpu_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15141807/cpu_add_run-config-lpar_mode-dedicated.log)
[cpu_rm_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15141808/cpu_rm_run-config-lpar_mode-dedicated.log)
[mem_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15141809/mem_add_run-config-lpar_mode-dedicated.log)

**Comment by PraveenPenguin** [comment]
@SamirMulani  can you please review this PR

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by Naresh-ibm** [comment]
Please add the doc string and remove the commented lines

**Comment by shirishaganta1** [comment]
JOB ID     : a477bac9653937c72544848871461c5f58d70f70
JOB LOG    : /home/siri/avocado-fvt-wrapper/results/job-2024-05-08T22.19-a477bac/job.log
 (1/6) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-9e77: STARTED
 (1/6) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-9e77:  PASS (675.36 s)
 (2/6) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-9e77: STARTED
 (2/6) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-9e77:  PASS (348.84 s)
 (3/6) dlpar_main.py:DlparTests.test_mix_cpu;run-config-lpar_mode-dedicated-9e77: STARTED
 (3/6) dlpar_main.py:DlparTests.test_mix_cpu;run-config-lpar_mode-dedicated-9e77:  PASS (781.38 s)
 (4/6) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-9e77: STARTED
 (4/6) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-9e77:  PASS (717.22 s)
 (5/6) dlpar_main.py:DlparTests.test_mem_rem;run-config-lpar_mode-dedicated-9e77: STARTED
 (5/6) dlpar_main.py:DlparTests.test_mem_rem;run-config-lpar_mode-dedicated-9e77:  PASS (597.37 s)
 (6/6) dlpar_main.py:DlparTests.test_mem_mix;run-config-lpar_mode-dedicated-9e77: STARTED
 (6/6) dlpar_main.py:DlparTests.test_mem_mix;run-config-lpar_mode-dedicated-9e77:  PASS (1186.37 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/siri/avocado-fvt-wrapper/results/job-2024-05-08T22.19-a477bac/results.html
JOB TIME   : 4334.11 s

File: dlpar/dlpar_api/api.py [hunk 1]
```diff
@@ -646,30 +656,40 @@ def __init__(self, config_payload, log='dedicated_cpu.log'):
...
         self.sleep_time = config_payload.get('sleep_time')
 
-        self.log.check_log('Getting Test configuration.',
-                           (self.quant_to_test is not None))
-        self.log.debug("Testing with %s Dedicated CPU units." %
-                       self.quant_to_test)
+        # self.log.check_log('Getting Test configuration.',
+        #                  (self.quant_to_test is not None))
+        # self.log.debug("Testing with %s Dedicated CPU units." %
+        #              self.quant_to_test)
```
[the remark this answers is not in the document]
Comment (shirishaganta1) [reply dlpar/dlpar_api/api.py:667]: we need them for move tests.. will uncomment them

File: dlpar/dlpar_api/api.py [hunk 2]
```diff
@@ -1058,11 +1078,11 @@ def __init__(self, config_payload, log='memory.log'):
...
 
-        self.log.debug("Testing with %s megabytes of memory." %
-                       self.quant_to_test)
+        # self.log.debug("Testing with %s megabytes of memory." %
+        #              self.quant_to_test)
         c_msg = 'Getting Test configuration.'
-        c_condition = self.quant_to_test is not None
-        self.log.check_log(c_msg, c_condition)
+        # c_condition = self.quant_to_test is not None
+        # self.log.check_log(c_msg, c_condition)
```
[the remark this answers is not in the document]
Comment (shirishaganta1) [reply dlpar/dlpar_api/api.py:1085]: we need them for move tests.. will uncomment them

**Comment by shirishaganta1** [comment]
shared:
JOB ID     : f749a165995fd831f8d7c5a4f08ba8dd56f85e22
JOB LOG    : /home/geet/avocado-fvt-wrapper/results/job-2024-05-14T23.47-f749a16/job.log
 (1/8) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-shared-a2b8: STARTED
 (1/8) dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-shared-a2b8:  PASS (2073.67 s)
 (2/8) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-shared-a2b8: STARTED
 (2/8) dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-shared-a2b8:  PASS (1956.62 s)
 (3/8) dlpar_main.py:DlparTests.test_mix_cpu;run-config-lpar_mode-shared-a2b8: STARTED
 (3/8) dlpar_main.py:DlparTests.test_mix_cpu;run-config-lpar_mode-shared-a2b8:  PASS (3769.27 s)
 (4/8) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-shared-a2b8: STARTED
 (4/8) dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-shared-a2b8:  PASS (648.03 s)
 (5/8) dlpar_main.py:DlparTests.test_mem_rem;run-config-lpar_mode-shared-a2b8: STARTED
 (5/8) dlpar_main.py:DlparTests.test_mem_rem;run-config-lpar_mode-shared-a2b8:  PASS (774.30 s)
 (6/8) dlpar_main.py:DlparTests.test_mem_mix;run-config-lpar_mode-shared-a2b8: STARTED
 (6/8) dlpar_main.py:DlparTests.test_mem_mix;run-config-lpar_mode-shared-a2b8:  PASS (1657.82 s)
 (7/8) dlpar_main.py:DlparTests.test_cpu_move;run-config-lpar_mode-shared-a2b8: STARTED
 (7/8) dlpar_main.py:DlparTests.test_cpu_move;run-config-lpar_mode-shared-a2b8:  PASS (63.70 s)
 (8/8) dlpar_main.py:DlparTests.test_mem_mov;run-config-lpar_mode-shared-a2b8: STARTED
 (8/8) dlpar_main.py:DlparTests.test_mem_mov;run-config-lpar_mode-shared-a2b8:  PASS (73.09 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/geet/avocado-fvt-wrapper/results/job-2024-05-14T23.47-f749a16/results.html
JOB TIME   : 11062.70 s

**Comment by shirishaganta1** [comment]
JOB ID     : 37b86c37d7bd484d0118896cef628fdb9fc90d2b
JOB LOG    : /home/siri/avocado-fvt-wrapper/results/job-2024-05-22T08.54-37b86c3/job.log
 (1/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-da08: STARTED
 (1/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_cpu_add;run-config-lpar_mode-dedicated-da08:  PASS (768.41 s)
 (2/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-da08: STARTED
 (2/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_cpu_rm;run-config-lpar_mode-dedicated-da08:  PASS (412.76 s)
 (3/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mix_cpu;run-config-lpar_mode-dedicated-da08: STARTED
 (3/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mix_cpu;run-config-lpar_mode-dedicated-da08:  PASS (1863.02 s)
 (4/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-da08: STARTED
 (4/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mem_add;run-config-lpar_mode-dedicated-da08:  PASS (538.30 s)
 (5/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mem_rem;run-config-lpar_mode-dedicated-da08: STARTED
 (5/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mem_rem;run-config-lpar_mode-dedicated-da08:  PASS (561.08 s)
 (6/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mem_mix;run-config-lpar_mode-dedicated-da08: STARTED
 (6/6) /home/siri/avocado-fvt-wrapper/tests/avocado-misc-tests/dlpar/dlpar_main.py:DlparTests.test_mem_mix;run-config-lpar_mode-dedicated-da08:  PASS (1260.88 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/siri/avocado-fvt-wrapper/results/job-2024-05-22T08.54-37b86c3/results.html
JOB TIME   : 5515.82 s
10:26:56 INFO    :

**Comment by shirishaganta1** [comment]
[test_cpu_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411515/test_cpu_add_run-config-lpar_mode-dedicated.log)
[test_cpu_rm_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411518/test_cpu_rm_run-config-lpar_mode-dedicated.log)
[test_mem_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411519/test_mem_add_run-config-lpar_mode-dedicated.log)
[test_mem_mix_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411523/test_mem_mix_run-config-lpar_mode-dedicated.log)
[test_mem_rem_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411524/test_mem_rem_run-config-lpar_mode-dedicated.log)
[test_cpu_add_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411528/test_cpu_add_run-config-lpar_mode-dedicated.log)
[test_mix_cpu_run-config-lpar_mode-dedicated.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/15411540/test_mix_cpu_run-config-lpar_mode-dedicated.log)

File: dlpar/dlpar_api/api.py [hunk 3]
```diff
@@ -452,6 +452,16 @@ def get_mem_option(self, linux_machine, option):
...
 
+    def get_lmb_value(self, linux_machine, option):
+        """to get lmb size of a managed system"""
+        o_cmd = 'lshwres -r mem -m ' + linux_machine.machine + \
+                ' --level sys -F ' + option
+        opt_value = self.hmc.sshcnx.cmd(o_cmd).stdout_text.strip()
+        d_msg = option + ": " + opt_value + " for CEC " + \
+            linux_machine.machine
+        self.log.debug(d_msg)
+        return opt_value
```
[the remark this answers is not in the document]
Comment (Naresh-ibm) [reply dlpar/dlpar_api/api.py:460]: @shirishaganta1 what is your thought on this?
@PraveenPenguin @abdhaleegit ^^

[the remark this answers is not in the document]
Comment (PraveenPenguin) [reply dlpar/dlpar_api/api.py:460]: @Naresh-ibm yes looks good idea, let us pursue this as enhancement as other API code can be considered , for now let go with current approach

[the remark this answers is not in the document]
Comment (Naresh-ibm) [reply dlpar/dlpar_api/api.py:460]: @PraveenPenguin ok. please approve if no more comments, will merge.

**Review by PraveenPenguin — APPROVED** [review]

**Review by Naresh-ibm — COMMENTED** [review]
@shirishaganta1 Thanks for PR

**Review by Naresh-ibm — APPROVED** [review]
LGTM

**Pull request merged**